### DISCUSSION TITLE
Just added a feature where user can select the tech stack of the zip-file they are uploading for doxifying/refactoring/tests.

### DISCUSSION THREAD

**Discussion by Abhishek891-beep** [lpelabs/reDocs.ai#15] [Show and tell]
I am just putting here to know if it is a good enough contribution to create a pull request .

![image](https://github.com/user-attachments/assets/38f5284a-ab67-43fc-a97a-70bae965debd)

![image](https://github.com/user-attachments/assets/2ebc25ab-6757-47ed-9f67-ead0fc84c8b3)

**Comment by HridayM25** [comment]
This seems really nice from the frontend side. Can you look into how we can change the functionality from the ML perspective? If a user selects a particular tech stack, you have that extra information you can use while prompting/ some other function. 
Can you even think if this can be automated? i.e how Github evaluates % of the stack based on the extension of the file; then the user need not even enter this manually.

**Reply by Abhishek891-beep** [reply]
Yeah , Github does this automatically , I think we need to add some logic in traverse_File function which goes through every folder of directory and could check the type of file also. 

Will look into the code again and see what i can do .

**Reply by HridayM25** [reply]
Yes, you can keep a dictionary ({"file_type":count}) when you traverse the file. You can then display the % of files in the frontend as a starting point.
